### PR TITLE
[rsm-binary-io] Add new port

### PR DESCRIPTION
**Pull request by Ryan-rsm-McKenzie**
**Describe the pull request**

- #### What does your PR fix?  
  Adds a new port [binary_io](https://github.com/Ryan-rsm-McKenzie/binary_io)

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  Supports only Linux/Windows + static linkage, and requires at least GCC 10

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -1364,6 +1364,8 @@ rsasynccpp:arm-uwp=fail
 rsasynccpp:x64-linux=fail
 rsasynccpp:x64-osx=fail
 # Requires g++10 but CI compiler only has g++9
```
Comment (Mrgreen2017) [inline scripts/ci.baseline.txt:1366]: Udididi

File: scripts/ci.baseline.txt [hunk 2]
```diff
@@ -1364,6 +1364,8 @@ rsasynccpp:arm-uwp=fail
...
+rsm-binary-io:x64-linux=fail
+# Requires g++10 but CI compiler only has g++9
```
Comment (PhoebeHui) [inline scripts/ci.baseline.txt:1368]:
```suggestion
```

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
@Ryan-rsm-McKenzie, thanks for your PR!

Could you change the 'tab' to 'space' in portfile.cmake?

File: scripts/ci.baseline.txt (hunk 2 above)
Comment (Ryan-rsm-McKenzie) [reply scripts/ci.baseline.txt:1368]: That second comment is for another port which also needs at least gcc 10

Comment (PhoebeHui) [reply scripts/ci.baseline.txt:1368]: I see, thanks!

**Review by PhoebeHui — APPROVED** [review]
LGTM, thanks for your updates!

**Comment by BillyONeal** [comment]
Thanks for the new port!

**Pull request merged**